extensions/workloads/create.go

```diff
@@ -18,20 +24,50 @@ func CreateDeploymentWithService(steveclient *steveV1.Client, wlName string, dep
 	logrus.Infof("Creating deployment: %s", wlName)
 	deploymentResp, err := steveclient.SteveType(DeploymentSteveType).Create(deployment)
 	if err != nil {
-		logrus.Errorf("Failed to create deployment: %s", wlName)
 		return nil, nil, err
 	}
 
-	logrus.Infof("Successfully created deployment: %s", wlName)
+	err = kwait.PollUntilContextTimeout(context.TODO(), 500*time.Millisecond, defaults.OneMinuteTimeout, true, func(ctx context.Context) (done bool, err error) {
+		deployment, err := steveclient.SteveType(DeploymentSteveType).ByID(deploymentResp.ID)
+		if err != nil {
+			return false, nil
+		}
+
+		if deployment.State.Name == active {
+			logrus.Infof("Successfully created deployment: %s", wlName)
+
+			return true, nil
+		}
+
+		return false, nil
+	})
+	if err != nil {
+		return nil, nil, err
+	}
 
 	logrus.Infof("Creating service: %s", service.Name)
 	serviceResp, err := steveclient.SteveType(ServiceType).Create(service)
 	if err != nil {
-		logrus.Errorf("Failed to create service: %s", service.Name)
 		return nil, nil, err
 	}
 
-	logrus.Infof("Successfully created service: %s", service.Name)
+	err = kwait.PollUntilContextTimeout(context.TODO(), 500*time.Millisecond, defaults.OneMinuteTimeout, true, func(ctx context.Context) (done bool, err error) {
+		service, err := steveclient.SteveType(ServiceType).ByID(serviceResp.ID)
+		if err != nil {
+			return false, nil
+		}
+
+		if service.State.Name == active {
+			logrus.Infof("Successfully created service: %s", service.Name)
+
+			return true, nil
+		}
+
+		return false, nil
+	})
+	if err != nil {
+		return nil, nil, err
+	}
```

Comment:
same comment here for waiting in a create call.

Reply:
Discussed offline, resolved.